File: SDKs/art19.com/1.0.0/python/src/sdk/models/operations/get_media_assets_id_.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import mediaasset as shared_mediaasset
from ..shared import resourcelink as shared_resourcelink


@dataclasses.dataclass
class GetMediaAssetsIDPathParams:
    id: str = dataclasses.field(metadata={'path_param': { 'field_name': 'id', 'style': 'simple', 'explode': False }})
    

@dataclass_json
@dataclasses.dataclass
class GetMediaAssetsID200ApplicationVndAPIPlusJSON:
    data: Optional[shared_mediaasset.MediaAsset] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('data') }})
    links: Optional[shared_resourcelink.ResourceLink] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('links') }})
    

@dataclasses.dataclass
class GetMediaAssetsIDRequest:
    path_params: GetMediaAssetsIDPathParams = dataclasses.field()
    

@dataclasses.dataclass
class GetMediaAssetsIDResponse:
    content_type: str = dataclasses.field()
    headers: dict[str, list[str]] = dataclasses.field()
    status_code: int = dataclasses.field()
    get_media_assets_id_200_application_vnd_api_plus_json_object: Optional[GetMediaAssetsID200ApplicationVndAPIPlusJSON] = dataclasses.field(default=None)
    

File: SDKs/art19.com/1.0.0/typescript/src/sdk/models/operations/getmediaassetsid.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetMediaAssetsIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=id" })
  id: string;
}


export class GetMediaAssetsId200ApplicationVndApiPlusJson extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=data" })
  data?: shared.MediaAsset;

  @SpeakeasyMetadata({ data: "json, name=links" })
  links?: shared.ResourceLink;
}


export class GetMediaAssetsIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetMediaAssetsIdPathParams;
}


export class GetMediaAssetsIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  headers: Record<string, string[]>;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  getMediaAssetsId200ApplicationVndApiPlusJsonObject?: GetMediaAssetsId200ApplicationVndApiPlusJson;
}
